spec/features/pets/12user_can_delete_pet_spec.rb

```diff
@@ -77,4 +77,160 @@
+    PetApp.create(pet: pet100, app: app111)
+    PetApp.create(pet: pet110, app: app100)
+
+    pet_ids = Pet.all.map{|pet| pet[:id]}
+    expect(pet_ids.include?(pet100.id)).to eq(true)
+    expect(pet_ids.include?(pet110.id)).to eq(true)
+    expect(pet_ids.include?(pet111.id)).to eq(true)
+
+    visit "/pets"
+
+    within("#pet-#{pet100.id}")do
+      expect(pet100.adoptable).to eq(false)
+      expect(page).to have_content("Cannot delete pet with approved applications")
+      expect(page).not_to have_button("Delete #{pet100.name}")
+    end
+
+    pet_ids = Pet.all.map{|pet| pet[:id]}
+    expect(pet_ids.include?(pet100.id)).to eq(true)
+
+    within("#pet-#{pet110.id}")do
+    expect(pet110.adoptable).to eq(true)
+      expect(page).not_to have_content("Cannot delete pet with approved applications")
+      click_button("Delete #{pet110.name}")
+    end
+
+    pet_ids = Pet.all.map{|pet| pet[:id]}
+    expect(pet_ids.include?(pet110.id)).to eq(false)
+
+    within("#pet-#{pet111.id}")do
+      expect(pet111.adoptable).to eq(true)
+      expect(page).not_to have_content("Cannot delete pet with approved applications")
+      click_button("Delete #{pet111.name}")
+    end
+
+    pet_ids = Pet.all.map{|pet| pet[:id]}
+    expect(pet_ids.include?(pet100.id)).to eq(true)
+    expect(pet_ids.include?(pet110.id)).to eq(false)
+    expect(pet_ids.include?(pet111.id)).to eq(false)
+
+  end
+
+  it "deleting pets pets deletes them from favorites" do
+    shelter1 = Shelter.create(name: "S1", address: "A1",city: "C1",state: "ST",zip: "12345")
+
+    pet100x = shelter1.pets.create(name: "P100x",age: "1",sex: "male",description: "d1",image: "i1",adoptable: false)
+    pet110x = shelter1.pets.create(name: "P110x",age: "1",sex: "male",description: "d1",image: "i1",adoptable: true)
+    pet111x = shelter1.pets.create(name: "P111x",age: "1",sex: "male",description: "d1",image: "i1",adoptable: true)
+
+    pet_ids = Pet.all.map{|pet| pet[:id]}
+    expect(pet_ids.include?(pet100x.id)).to eq(true)
+    expect(pet_ids.include?(pet110x.id)).to eq(true)
+    expect(pet_ids.include?(pet111x.id)).to eq(true)
+
+    visit "/pets"
+    expect(page).to have_css(".favorite_indicator", text: 0)
+    within("#fav-#{pet111x.id}") do
+      click_button "Add #{pet111x.name} to favorites"
+    end
+    expect(page).to have_content("You have added #{pet111x.name} to your favorites")
+    expect(page).to have_css(".favorite_indicator", text: 1)
+
+    within("#fav-#{pet100x.id}") do
+      click_button "Add #{pet100x.name} to favorites"
+    end
+    expect(page).to have_content("You have added #{pet100x.name} to your favorites")
+    expect(page).to have_css(".favorite_indicator", text: 2)
+
+    expect(current_path).to eq("/pets")
+    within("#pet-#{pet111x.id}")do
+      within(".delete-form")do
+        click_button("Delete P111x")
+      end
+    end
+    pet_ids = Pet.all.map{|pet| pet[:id]}
+    expect(pet_ids.include?(pet100x.id)).to eq(true)
```

Comment:
this is a way I found to check if a pet has been deleted. Open to suggestions for improvement